pkg/query/path_test.go

```diff
@@ -651,3 +651,405 @@ func TestPath_ConversionRoundtrip(t *testing.T) {
 		require.Equal(originalRel.OptionalIntegrity, convertedRel.OptionalIntegrity)
 	})
 }
+
+func TestPath_EqualsEndpoints(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	t.Run("nil_paths", func(t *testing.T) {
+		t.Parallel()
+
+		var path1, path2 *Path
+		require.True(path1.EqualsEndpoints(path2), "both nil should be equal")
+
+		path1 = &Path{Resource: NewObject("document", "doc1")}
+		require.False(path1.EqualsEndpoints(path2), "nil and non-nil should not be equal")
+		require.False(path2.EqualsEndpoints(path1), "non-nil and nil should not be equal")
+	})
+
+	t.Run("same_endpoints", func(t *testing.T) {
+		t.Parallel()
+
+		path1 := &Path{
+			Resource: NewObject("document", "doc1"),
+			Relation: "viewer", // Relations don't matter for endpoint comparison
+			Subject:  NewObjectAndRelation("alice", "user", "..."),
+		}
+
+		path2 := &Path{
+			Resource: NewObject("document", "doc1"),
+			Relation: "editor",                                        // Different relation but same endpoints
+			Subject:  NewObjectAndRelation("alice", "user", "member"), // Different subject relation but same endpoint
```

Comment:
The subject relation being different but being the same endpoints is kinda surprising to me - it feels like a subject relation is pointing off and away, and so it shouldn't compare as equal, but I'm assuming that's because of a gap in understanding.